go/vt/vtgate/plugin_mysql_server.go

```diff
@@ -20,9 +22,46 @@ import (
 )
 
 var (
-	mysqlServerPort = flag.Int("mysql_server_port", 0, "If set, also listen for MySQL binary protocol connections on this port.")
+	mysqlServerPort               = flag.Int("mysql_server_port", 0, "If set, also listen for MySQL binary protocol connections on this port.")
+	mysqlAuthServerImpl           = flag.String("mysql_auth_server_impl", "config", "Which auth server implementation to use.")
+	mysqlAuthServerConfigFile     = flag.String("mysql_auth_server_config_file", "", "JSON File to read the users/passwords from.")
+	mysqlAuthServerConfigString   = flag.String("mysql_auth_server_config_string", "", "JSON representation of the users/passwords config.")
+	mysqlAllowClearTextWithoutTLS = flag.Bool("mysql_allow_clear_text_without_tls", false, "If set, the server will allow the use of a clear text password over non-SSL connections.")
 )
 
+// Handles initializing the AuthServerConfig if necessary.
+func initAuthServerConfig() {
```

Comment:
I didn't want to have the flags defined in the library, as then they would end up being in all of our binaries (like vttablet) when they're unused. But I see your point, will move the init code itself in the library.